### ISSUE TITLE
Improve Documentation with Examples

### ISSUE THREAD

**Issue by iliaMonfared** [ChadSki/SharpNeedle#2]
Hi,
Can you explain more how I should inject a dll into a running process? And how can I run a method from dll?
Thanks.

**Comment by marcussacana** [comment]
i like know if is possible "hook/proxy" using C# + this project... for sample, in my case, i need proxy the GDI32 => CreateFontIndirectW

**Comment by ChadSki** [comment]
No, this project is for injecting a DLL that does arbitrary things. It doesn't provide a general way of hooking functions. You could look into the Microsoft project "Detours", or one of the many alternatives.

**Comment by kkcosmo** [comment]
Could you add some more explanation for the failing of the injection starting always with  CallExport (from the Injection.cpp), e.g.:
- "Could not create thread in remote process."
- "Could not get handle to process."

Providing more details in why they happen and what could be done to change that.

Also, do you know of a way to get access to all processes (besides providing admin rights, which wont do the trick for me). I get error code 5 for access denied on certain processes.

**Comment by kkcosmo** [comment]
Running the Launcher as SYSTEM did do the trick for the access rights, still I get
"CallExport: Could not get module Snapshot for remote process."
